### PR TITLE
Update AGP to 7.3.0, java to 11, kotlin to 1.6.10

### LINKED ISSUE #373: Update example to target Android 13
### Description

If the example would be updated to compile/targetsdk 33, it would show that Sentry is working fine with apps that target the new SDK.

### PR DESCRIPTION
**Pull request by romtsn**
## :scroll: Description

* Bumps current AGP version to 7.3.0
* Bumps preview AGP versions to 7.4.0-beta01 and 8.0.0-alpha01
* Bumps kotlin to 1.6.10, languageVersion to 1.4 (1.4 is supported from Gradle 7.0+, which is what our plugin supports anyway)
* Bumps Java to 11 (our plugin supports only AGP 7.0+, which, in turn, supports java 11+)
* Bumps compileSdkVersion to 33 on samples (tested manually, no problems)
* Switch from AndroidManifest package name to `namespace` attribute in gradle dsl

## :bulb: Motivation and Context

Closes #373 #364 #274

## :pencil: Checklist

- [x] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] No breaking changes

### CODE REVIEW

File: plugin-build/build.gradle.kts [hunk 1]
```diff
@@ -62,27 +62,28 @@ dependencies {
 }
 
 configure<JavaPluginExtension> {
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(11))
```
Comment (romtsn) [inline plugin-build/build.gradle.kts:66]: @marandaneto do you see any problems with that? We support AGP 7.0 min, which in turn supports Java 11 as min version, so I think we are good

Comment (marandaneto) [reply plugin-build/build.gradle.kts:66]: It should be fine.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+- Bump AGP to v7.3.0
```
Comment (romtsn) [inline CHANGELOG.md:5]:
```suggestion
- Bump AGP to v7.3.0 ([#378](https://github.com/getsentry/sentry-android-gradle-plugin/pull/378))
```

File: buildSrc/src/main/java/Dependencies.kt [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
+    const val DOKKA = "1.6.10"
+    const val KOTLIN = "1.6.10"
     const val AAR_2_JAR = "0.6"
     const val KTLINT = "10.2.1"
     // do not upgrade to 0.18.0, it does not generate the pom-default.xml and module.json under
     // build/publications/maven
     const val MAVEN_PUBLISH = "0.17.0"
     const val PROGUARD = "7.1.0"
+
+    fun isProguardApplicable(): Boolean = VersionNumber.parse(AGP).major < 8
```
Comment (romtsn) [inline buildSrc/src/main/java/Dependencies.kt:14]:
```suggestion
	
	// proguard does not support AGP 8 yet
    fun isProguardApplicable(): Boolean = VersionNumber.parse(AGP).major < 8
```

**Comment by marandaneto** [comment]
@romtsn you can likely close https://github.com/getsentry/sentry-android-gradle-plugin/issues/274 as well

**Comment by romtsn** [comment]
@marandaneto oh yeah, thanks for pointing out

File: plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt [hunk 4]
```diff
@@ -94,18 +94,15 @@ abstract class BaseSentryPluginTest(
...
-            arrayOf("7.1.3", "7.3.3"),
-            arrayOf("7.1.3", "7.4"),
-            arrayOf("7.1.3", "7.5"),
             arrayOf("7.2.1", "7.3.3"),
             arrayOf("7.2.1", "7.4"),
             arrayOf("7.2.1", "7.5"),
-            arrayOf("7.3.0-beta05", "7.5"),
-            arrayOf("7.4.0-alpha03", "7.5")
+            arrayOf("7.3.0", "7.4"),
+            arrayOf("7.3.0", "7.5"),
```
Comment (adinauer) [inline plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt:103]: Should we change `7.5` to `7.5.1` for gradle?

**Review by adinauer — APPROVED** [review]
LGTM; one question regarding gradle version

File: plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt (hunk 4 above)
Comment (romtsn) [reply plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt:103]: they don't do breaking api/behavior changes in patch versions, so we should be good here I think

**Pull request merged**